src/componentTypesView.ts

```diff
@@ -4,52 +4,38 @@
  *-----------------------------------------------------------------------------------------------*/
 
 import {
-    TreeDataProvider,
     TreeItem,
-    Event,
-    EventEmitter,
     TreeView,
     window,
     Uri,
-    TreeItemCollapsibleState,
     commands,
+    workspace,
+    ProviderResult,
 } from 'vscode';
-import * as path from 'path';
-import { CliExitData } from './cli';
 import { getInstance, Odo, OdoImpl } from './odo';
-import { Command } from './odo/command';
 import {
-    ComponentTypeDescription,
-    ComponentTypesJson,
-    DevfileComponentType,
-    isDevfileComponent,
-    isRegistry,
     Registry,
 } from './odo/componentType';
-import { isStarterProject, StarterProject } from './odo/componentTypeDescription';
+import { StarterProject } from './odo/componentTypeDescription';
 import { vsCommand, VsCommandError } from './vscommand';
-import { Platform } from './util/platform';
 import validator from 'validator';
+import { BaseTreeDataProvider } from './base/baseTreeDataProvider';
+import { WorkspaceEntry, WorkspaceFolderComponent } from './componentsView';
+import { Application } from 'express';
+import { Project } from './odo/project';
 
-type ComponentType = DevfileComponentType | StarterProject | Registry;
+type ComponentType = Project | Application | WorkspaceEntry | WorkspaceFolderComponent;
```

Comment:
This is registry tree view. It has nothing to do with OpenShift cluster. Projects, Applications and etc. do not belong to this view.